### PR TITLE
Card SubLinks

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This adds support for sublinks in cards

Screenshot 2022-04-25 at 10 00 12
Screenshot 2022-04-25 at 10 00 36

### Still pending
The styling for sublinks needs to vary based on the format of the _container_ that they exist in

Here we see links to standard news articles looking broken because they exist in a card with the `LiveBlog` format
Screenshot 2022-04-25 at 09 59 48

Here we see a link to a liveblog looking broken because the container is standard
Screenshot 2022-04-25 at 09 59 32

The work to vary styles based on container type / palette is pending and will be completed in a different PR

### `66/34` vs. `67/33`
I've slightly adjusted the positioning of content when a card is horizontal by changing the percentages used by one point. I did this because, 1) this aligns us with Frontend and 2) this means the sublinks will vertically align with the image now
Screenshot 2022-04-25 at 10 07 26

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
Nice addition.

A thought regarding the percentages: we decided to avoid `quarter`, `third`, `half`, `two-thirds`, `three-quarters` and `full` so as not to create an unnecessary abstraction. There might be small differences in how browsers decide to render these values anyways, so if we actually want pixel-perfection everywhere, we probably would need some new kind of abstraction.

**Comment by oliverlloyd** [comment]
> we decided to avoid

I wasn't aware of this but it definitely sounds like something worth investigating 👍

File: dotcom-rendering/src/web/components/Card/Card.tsx [hunk 1]
```diff
@@ -157,6 +160,84 @@ export const Card = ({
 
 	const cardPalette = decidePalette(format);
 
+	const isHorizontal = imagePosition === 'left' || imagePosition === 'right';
+	const moreThanTwoSubLinks =
+		supportingContent &&
+		supportingContent.length &&
+		supportingContent.length > 2;
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Card/Card.tsx:166]: Thanks!

**Comment by mxdvl** [comment]
> > we decided to avoid
> 
> I wasn't aware of this but it definitely sounds like something worth investigating 👍

I’m assuming this was the case, actually, I should have written: “I assume we decided to avoid abstractions like `quarter`, etc.” Finding the right balance of abstraction is a constant exercise, and I don’t think small refactors like this would be reason enough to introduce it. Again, we have Chromatic, so we can feel confident!

File: dotcom-rendering/src/web/components/SupportingContent.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+const directionStyles = (imagePosition: ImagePositionType) => {
+	switch (imagePosition) {
+		case 'left':
+		case 'right':
+			return css`
+				flex-direction: column;
+				${from.phablet} {
+					flex-direction: row;
+				}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/SupportingContent.tsx:26]: Looking at Frontend, this should have switched from the `tablet` breakpoint.

Fixed in [ec72534](https://github.com/guardian/dotcom-rendering/pull/6598/commits/ec72534d72b91c11a6a838e96ed0cb152a81d594)

**Pull request merged**